src/cp.js

```diff
@@ -72,10 +74,13 @@ function copyFileSync(srcFile, destFile, options) {
       pos += bytesRead;
     }
 
+    if (options.preserve) {
+      fs.fchownSync(fdw, srcStat.uid, srcStat.gid);
+      fs.futimesSync(fdw, srcStat.atime, srcStat.mtime);
```

Comment:
On linux, `fs.utimesSync()` always produces timestamps rounded off to `seconds` but original files could have  timestamps of millisecond level. But I don't why is that so I probably need more time to figure it out.
However `fs.futimesSync()` could produce timestamps of millisecond level. So I put it this way for now.

Also I tried to add sleep in test but it didn't work.

Reply:
Huh, ok. This implementation is fine. I will add a comment explaining that `utimesSync` doesn't work but `futimesSync` does.